src/overlays/actors/ovl_En_Ot/z_en_ot.c

```diff
@@ -41,128 +81,1101 @@ const ActorInit En_Ot_InitVars = {
+                    break;
+            }
+            break;
+
+        case 2:
+            D_80B5E888 = this;
+
+            switch (this->unk_344) {
+                case 0:
+                    Actor_SetScale(&this->actor, 0.0f);
+                    if (!(gSaveContext.weekEventReg[13] & 1)) {
+                        Actor_SetScale(&this->actor, 0.0f);
+                        func_80B5C910(this, globalCtx);
+                    } else {
+                        Player* player = PLAYER;
+
+                        if (func_800C9B90(&globalCtx->colCtx, player->actor.floorPoly, player->actor.floorBgId)) {
+                            Actor_SetScale(&this->actor, 0.0f);
+                            func_80B5C910(this, globalCtx);
+                        } else {
+                            this->unk_360 = D_80B5E884;
+                            this->unk_360->unk_360 = this;
+                            func_80B5C9D0(this, globalCtx);
+                        }
+                    }
+                    break;
+
+                case 1:
+                    Actor_SetScale(&this->actor, 0.012999999f);
+                    if (gSaveContext.weekEventReg[84] & 0x10) {
+                        if (gSaveContext.weekEventReg[32] & 1) {
+                            func_80B5C244(this, globalCtx);
+                        } else {
+                            func_80B5C684(this, globalCtx);
+                        }
+                    } else {
+                        func_80B5CE6C(this, globalCtx);
+                    }
+                    break;
+            }
+            break;
+
+        case 3:
+            if (!(gSaveContext.weekEventReg[26] & 8)) {
+                this->actor.flags |= 0x8000000;
+                this->actor.flags &= ~(0x8 | 0x1);
+                Actor_SetScale(&this->actor, 0.0064999997f);
+                this->collider.dim.radius *= 0.5f;
+                this->collider.dim.height *= 0.5f;
+                this->collider.dim.yShift *= 0.5f;
+                this->actor.update = func_80B5DAEC;
+                func_80B5C634(this, globalCtx);
+            } else {
+                Actor_MarkForDeath(&this->actor);
+            }
+            break;
+    }
+}
+
+void EnOt_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnOt* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+    LightContext_RemoveLight(globalCtx, &globalCtx->lightCtx, this->lightNode);
+}
+
+void func_80B5BAAC(LightInfo* lightInfo, Vec3f* arg1, Color_RGB8* arg2, s16 radius) {
+    Lights_PointNoGlowSetInfo(lightInfo, arg1->x, arg1->y, arg1->z, arg2->r, arg2->g, arg2->b, radius);
+}
+
+void func_80B5BB38(Color_RGB8* arg0, Color_RGB8* arg1, f32 arg2) {
+    f32 rand = Rand_ZeroOne();
+
+    arg0->r = (arg1->r * arg2) + (arg1->r * (1.0f - arg2) * rand);
+    arg0->g = (arg1->g * arg2) + (arg1->g * (1.0f - arg2) * rand);
+    arg0->b = (arg1->b * arg2) + (arg1->b * (1.0f - arg2) * rand);
```

Comment:
```suggestion
    arg0->r = F32_LERP(arg1->r * rand, arg1->r, arg2);
    arg0->g = F32_LERP(arg1->g * rand, arg1->g arg2);
    arg0->b = F32_LERP(arg1->b * rand, arg1->b, arg2);
```
How much flexibility is there in reordering the terms without breaking regalloc? This would be equivalent but would also likely create non-matching. But its worth a shot

Reply:
Yeah doesn't match unfortunately.